packages/malloy-syntax-highlight/README.md

```diff
@@ -0,0 +1,120 @@
+
+```
+npm run gen-malloy-tokens
+```
+
+## Testing TextMate grammars
+
+This package's TextMate tokenization tests verify both the scope names (ex. _constant.numeric.date_, _keyword.control.select_) applied to patterns and the foreground text color applied by those styles.
+
+TextMate tests are written in TypeScript and Jest in files with the suffix `.spec.ts`. `grammars/malloy/malloy.spec.ts` provides an example test for the Malloy grammar using the Dark+ theme, including the config object for the test which can be found in `test/config/textmate/malloyDarkPlusConfig.ts`. You'll notice that this is the same config that is used to generate tokenizations artifacts above.
+
+To run all TextMate tests, use:
+
+```
+npm run test-textmate-grammars
+```
+
+## Testing Monarch grammars
+
+This package's Monarch tokenization tests verify parity between the foreground text color of lines highlighted using the TextMate grammar and its corresponding Monarch grammar.
+
+Monarch tests are written in TypeScript and Jasmine with the file suffix `.test.ts` and run in the browser to use the `monaco-editor` package's API, hence the use of Karma with the configuration in `karma.conf.js`. `grammars/malloy/malloy.test.ts` provides an example test for the Malloy grammar using the Dark+ theme, including the config object for the test which can be found in `test/config/monaco/malloyDarkPlusConfig.ts`.
+
+To run all Monarch tests, use:
+
+```
+npm run test-monarch-grammars
+```
+
+This command uses an `es6` TypeScript compilation target to generate JavaScript files in a temporary dist/ folder that are compatible with Karma's mechanisms for ES6 module support. The Karma tests will then be launched in Chrome: if Karma reports a failure in Chrome, you may need to refresh the page. Note that the dist/ folder is deleted before every run of `tsc` so you need not worry about Karma serving files that are out-of-date.
+
+## Generating a Monarch grammar
+
+In `scripts/generateMonarchGrammar.ts`, you will find a script that generates a Monarch syntax highlighting grammar from a TextMate grammar. In `package.json` you will find a script definition `gen-malloy-monarch` that runs this script for the Malloy TextMate grammar. The script's first command line argument is the input file to parse the TextMate specification from and the second command line argument is the output file to write the Monarch definition to.
+
+To visually inspect how well the script works out of box, run:
+
+```
+npm run gen-malloy-monarch
```

Comment:
Same formatting question as above, and same question about "this is a generated file".

Reply:
Sounds good, as for the mention of it being auto-generated, I'll also make a note that users can and will likely need to run lint-fix in from either the package directory or the root of the monorepo